Sources/Profile-View.php

```diff
@@ -451,24 +592,12 @@ function showAlerts($memID)
 	// Are we opening a specific alert ?
 	if (!empty($_REQUEST['alert']))
 	{
-		$alert_id = (int)$_REQUEST['alert'];
-		$request = $smcFunc['db_query']('', '
-			SELECT id_member, id_member_started, content_type, content_action, content_id, is_read, extra
-			FROM {db_prefix}user_alerts
-			WHERE id_alert = {int:alert}',
-			array(
-				'alert' => $alert_id
-			)
-		);
-		$alert = $smcFunc['db_fetch_assoc']($request);
-		$smcFunc['db_free_result']($request);
+		$alert_id = (int) $_REQUEST['alert'];
 
-		// Can the user see this alert ?
-		if($memID != $alert['id_member'])
-			redirectexit();
+		$alert = array_pop(fetch_alerts($memID, $alert_id));
```

Comment:
You should assign the result to a variable first and then use it in array_pop, as I am getting 
"**Only variables should be passed by reference**" in the errolog